cluster/charts/gitlab-controller/crds/controller/v1alpha1/gitlab.yaml

```diff
@@ -0,0 +1,85 @@
+apiVersion: apiextensions.k8s.io/v1beta1
+kind: CustomResourceDefinition
+metadata:
+  creationTimestamp: null
+  labels:
+    controller-tools.k8s.io: "1.0"
+  name: gitlabs.controller.gitlab.io
+spec:
+  additionalPrinterColumns:
+  - JSONPath: .status.state
+    name: STATE
+    type: string
+  - JSONPath: .status.endpoint
+    name: ENDPOINT
+    type: string
+  group: controller.gitlab.io
+  names:
+    kind: GitLab
+    plural: gitlabs
+  scope: Namespaced
+  subresources:
+    status: {}
+  validation:
+    openAPIV3Schema:
+      properties:
+        apiVersion:
+          description: 'APIVersion defines the versioned schema of this representation
+            of an object. Servers should convert recognized schemas to the latest
+            internal value, and may reject unrecognized values. More info: https://git.k8s.io/community/contributors/devel/api-conventions.md#resources'
+          type: string
+        kind:
+          description: 'Kind is a string value representing the REST resource this
+            object represents. Servers may infer this from the endpoint the client
+            submits requests to. Cannot be updated. In CamelCase. More info: https://git.k8s.io/community/contributors/devel/api-conventions.md#types-kinds'
+          type: string
+        metadata:
+          type: object
+        spec:
+          properties:
+            domain:
+              description: Domain which will contain records to resolve gitlab, registry,
+                and minio (if enabled) to the appropriate IP
+              type: string
+            email:
+              description: Email address to register TLS certificates
+              type: string
+            hostSuffix:
+              description: HostSuffix appended to domain records, i.e. `gitlab-suffix.domain`,
+                default - no suffix
+              type: string
+            port:
+              description: Port gitlab service port, default - none
+              type: uint
+            protocol:
+              description: Protocol http or https, default - https
+              type: string
+            providerRef:
+              description: ProviderRef cloud provider reference
+              type: object
+            reclaimPolicy:
+              description: ReclaimPolicy controls application cleanup
+              type: string
```

Comment:
For the immediate term before we have automatic ways of this getting applied to a cluster, it seems like we might be missing a reference to the cluster we wish to deploy our gitlab onto.

Reply:
Good point. The `KubernetesApplication` this creates has a scheduler which will pick an eligible cluster based on its `.spec.clusterSelector`, but we'll need to know how to set that selector.

Reply:
This is a good question. Ultimately, it boils down to - does the end user know about target clusters? If so, then yes, then the target cluster reference should be passed along, otherwise, the GitLab controller will have to do some "magic" trying to figure out how to assign cluster selector.

Reply:
I think the user would want to operate on an auto-scaler and it's policy/preferences, but long term having an optional override or a preference in case they want to make that decision doesn't seem bad either.

Reply:
Adding: @thephred - since this impacts his demo.

Reply:
@lukeweber @negz - in a similar vein: what about target namespace? Is it left entirely up to `GitLabControler` to decide which target namespace to propagate the GitLab application(s)

Reply:
@ichekrygin It feels appropriate to me in this context to all the user to specify the namespace in as `.spec.targetNamespace` or something.

I think there's two possible use cases here:
1. The user wants to deploy to a cluster that "does other things", by which I mean it's an existing cluster running non-Crossplane workloads, or an existing cluster running other Crossplane workloads.
1. The user just wants to deploy Gitlab and wants Crossplane to create a cluster for it - the cluster will never run anything but Gitlab. In this case I think we should just deploy to the `default` namespace.

So perhaps we should allow setting `.spec.targetNamespace` but have something (a webhook?) default that field to `default` if it's not specified?

Reply:
@negz - I think I like the idea of the default namespace. What is your take on `gitlab` vs. `default` namespace?

Reply:
@lukeweber - what is your take on additional user value: `targetNamespace`?

Reply:
@ichekrygin I think it make sense to use `default` _if_ we really are doubling down on our "opinionated" long term Crossplane vision - i.e. that clusters are cattle and we'll regularly be spinning up whole clusters to run only one complex workload. If the cluster's only purpose is to run Gitlab then I think Gitlab should just be in `default`.

In any other case, I think it's safer to use namespace `gitlab`.

Reply:
targetNamespace isn't a part of any demo scenario we have at the moment, but that being said we're early days. It's a bit speculative based on what we think we will schedule in a target cluster in crossplane. I can see the idea of configuring it could make sense, but a future scheduler could also always deploy a version of a workload in a unique namespace as a form of naive isolation. 

For me it kind of touches design issues that maybe have already been discussed on the crossplane side. What will we deploy in any given target cluster and how: a single workload, related workloads, and what's the upgrade path for a well behaved workload?